src/tun2socks/build.action.mjs

```diff
@@ -0,0 +1,130 @@
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import url from 'url';
+import os from 'node:os';
+import path from 'node:path';
+import fs from 'node:fs/promises';
+
+import {spawnStream} from '../build/spawn_stream.mjs';
+import {getBuildParameters} from '../build/get_build_parameters.mjs';
+
+/**
+ * @description Builds the tun2socks library for the specified platform.
+ *
+ * @param {string[]} parameters
+ */
+export async function main(...parameters) {
+  const {platform: targetPlatform} = getBuildParameters(parameters);
+
+  const binDir = path.join(process.env.OUTPUT_DIR, 'bin');
+  const buildDir = path.join(
+    process.env.BUILD_DIR,
+    ['ios', 'macos', 'maccatalyst'].includes(targetPlatform) ? 'apple' : targetPlatform
+  );
+
+  await fs.mkdir(binDir, {recursive: true});
+  await fs.mkdir(buildDir, {recursive: true});
+
+  // install go tools locally
+  await spawnStream(
+    'go',
+    'build',
+    '-o',
+    binDir,
+    'golang.org/x/mobile/cmd/gomobile',
+    'golang.org/x/mobile/cmd/gobind',
+    'github.com/crazy-max/xgo'
+  );
+
+  const hostPlatform = os.platform();
+
+  process.env.PATH = hostPlatform === 'win32' ? `${binDir};${process.env.PATH}` : `${binDir}:${process.env.PATH}`;
+
+  switch (targetPlatform + hostPlatform) {
+    case 'android' + 'darwin':
+    case 'android' + 'linux':
+    case 'android' + 'win32':
+      return spawnStream(
+        'gomobile',
+        'bind',
+        '-androidapi=33',
+        '-target=android',
+        `-o=${buildDir}/tun2socks.aar`,
+
+        'github.com/Jigsaw-Code/outline-client/src/tun2socks/outline/tun2socks',
+        'github.com/Jigsaw-Code/outline-client/src/tun2socks/outline/shadowsocks'
+      );
+    case 'ios' + 'darwin':
+    case 'macos' + 'darwin':
+    case 'maccatalyst' + 'darwin':
+      process.env.MACOSX_DEPLOYMENT_TARGET = '10.14';
+
+      return spawnStream(
+        'gomobile',
+        'bind',
+        '-bundleid=org.outline.tun2socks',
+        '-iosversion=13.1',
+        `-target=ios,iossimulator,macos,maccatalyst`,
+        `-o=${buildDir}/tun2socks.xcframework`,
+
+        'github.com/Jigsaw-Code/outline-client/src/tun2socks/outline/tun2socks',
+        'github.com/Jigsaw-Code/outline-client/src/tun2socks/outline/shadowsocks'
+      );
+    case 'windows' + 'linux':
+    case 'windows' + 'darwin':
+      await spawnStream(
+        'xgo',
+        '-targets=windows/386',
+        `-dest=${buildDir}`,
+
+        '-pkg=src/tun2socks/outline/electron',
+        '-out=tun2socks',
+        '.'
+      );
+
+      return spawnStream('cp', `${buildDir}/tun2socks-windows-386.exe`, `${buildDir}/tun2socks.exe`);
```

Comment:
Why don't we use `fs.copyFile` here (or `fs/promise`)? I think it's more cross-platform than `cp`.

Reply:
I see, maybe add a TODO in case anyone in the future would like to investigate into that?